pages/login/LoginHandler.php

```diff
@@ -75,7 +75,7 @@ public function index($args, $request)
         $templateMgr = TemplateManager::getManager($request);
         $templateMgr->assign([
             'loginMessage' => $request->getUserVar('loginMessage'),
-            'username' => $request->getUserVar('username') ?? $session->getSessionVar('username'),
+            'username' => $session->getSessionVar('email') ?? $session->getSessionVar('username'),
```

Comment:
@touhidurabir, can you explain this change? I'm not sure I'm following.

Reply:
@asmecher this is to prefill the auth key (username/email) with the email if one of last login done via email instead of username . So if user login using the email, then logout and if we only have 
```php
$request->getUserVar('username') ?? $session->getSessionVar('username')
```
the the login form is prefilled with username only even if their last login is via email . as once a login done via email, we are putting the email information in session , we can prefill it in the login form . 

I am happy that you raised this point . This lead me to think 2 more cases 
1. If we keep the changes that I have proposed, I think we need to update the behaviour a bit when a user update the email . Otherwise it will sill prefill the older/incorrect email address . So we need to update the session data on email update . 
2. As we are allowing email as auth key and which is possible to update unlike the `username`, I think we need to implement the session invalidation like we did at https://github.com/pkp/pkp-lib/issues/6983 . 

what you think above 2 points ?

Reply:
@asmecher implementation done based on above those 2 point .